packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,166 @@
+          delete cfnContent.Outputs.CloudWatchEventRule;
+          break;
+        }
+      }
+      fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+    }
+  }
+  return {
+    cloudwatchRule: scheduleParams.cloudwatchRule,
+  };
+}
+
+export async function cronServiceWalkthrough(context: any) {
+  let cloudwatchRule;
+  // resource questions for setting cron
+  const intervalQuestion = {
+    type: 'list',
+    name: 'interval',
+    message: 'At which interval should the function be invoked:',
+    choices: ['Minutes', 'Hourly', 'Daily', 'Weekly', 'Monthly', 'Yearly', 'Custom AWS cron expression'],
+  };
+  const intervalAnswer = await inquirer.prompt([intervalQuestion]);
+  switch (intervalAnswer.interval) {
+    case CronExpressionsMode.Minutes: {
+      cloudwatchRule = minuteHelper(context);
+      break;
+    }
+    case CronExpressionsMode.Hourly: {
+      cloudwatchRule = hourHelper(context);
+      break;
+    }
+    case CronExpressionsMode.Daily: {
+      var exp = new CronBuilder();
+      exp = await timeHelper(exp);
+      cloudwatchRule = exp.build();
+      cloudwatchRule = 'cron(' + replaceAt(cloudwatchRule, cloudwatchRule.lastIndexOf('*'), '?') + ' ' + '*' + ')';
+      break;
+    }
+    case CronExpressionsMode.Weekly: {
+      var exp1 = new CronBuilder();
+      exp1 = await weekHelper(exp1);
+      exp1 = await timeHelper(exp1);
+      cloudwatchRule = exp1.build();
+      cloudwatchRule = 'cron(' + cloudwatchRule + ' ' + '*' + ')';
+      break;
+    }
+    case CronExpressionsMode.Monthly: {
+      var exp2 = new CronBuilder();
+      exp2 = await monthHelper(exp2, context);
+      exp2 = await timeHelper(exp2);
+      cloudwatchRule = exp2.build();
+      cloudwatchRule = 'cron(' + replaceAt(cloudwatchRule, cloudwatchRule.lastIndexOf('*'), '?') + ' ' + '*' + ')';
+      break;
+    }
+    case CronExpressionsMode.Yearly: {
+      var exp3 = new CronBuilder();
+      exp3 = await yearHelper(exp3, context);
+      exp3 = await timeHelper(exp3);
+      cloudwatchRule = exp3.build();
+      cloudwatchRule = 'cron(' + replaceAt(cloudwatchRule, cloudwatchRule.lastIndexOf('*'), '?') + ' ' + '*' + ')';
+
+      break;
+    }
+    case CronExpressionsMode.CUSTOM: {
+      const customRuleQuestion = {
+        type: 'input',
+        name: 'customRule',
+        message: 'Custom Schedule expression(Learn more: https://amzn.to/3akXtJF)',
+        validate: ValidCronExpression({
+          onErrorMsg: 'Enter a valid Schedule Expression (Learn more: https://amzn.to/3akXtJF)',
+        }),
+      };
+      const customRuleAnswer = await inquirer.prompt([customRuleQuestion]);
+      cloudwatchRule = 'cron(' + customRuleAnswer.customRule + ')';
+
+      break;
+    }
+  }
+  return cloudwatchRule;
+}
+
+function replaceAt(string, index, replace) {
+  return string.substring(0, index) + replace + string.substring(index + 1);
+}
+
+function ValidCronExpression(validation) {
```

Comment:
I still think you should combine the functionality of this function and isValidCronExpression into one, but if not, at least rename this one to `validCronExpression`